src/rpc/format/cbor/convert.rs

```diff
@@ -59,6 +59,19 @@ impl TryFrom<Cbor> for Value {
 							Ok(v) => Ok(v.into()),
 							_ => Err("Expected a valid Datetime value"),
 						},
+						Data::Array(v) if v.len() == 2 => {
+							let seconds = match Value::try_from(Cbor(v[0].clone())) {
+								Ok(Value::Number(Number::Int(v))) => v,
+								_ => return Err("Expected a CBOR integer data type"),
+							};
+
+							let nanos = match Value::try_from(Cbor(v[1].clone())) {
+								Ok(Value::Number(Number::Int(v))) => v,
+								_ => return Err("Expected a CBOR integer data type"),
+							};
+
+							Utc.timestamp_opt(seconds, nanos)?.into()
+						}
```

Comment:
We should be able to do without the `clone()`s here. This is just off the top of my head, but perhaps the following works:

```suggestion
						Data::Array(mut v) if v.len() == 2 => {
							let seconds = match Value::try_from(Cbor(v.remove(0))) {
								Ok(Value::Number(Number::Int(v))) => v,
								_ => return Err("Expected a CBOR integer data type"),
							};

							let nanos = match Value::try_from(Cbor(v.remove(0))) {
								Ok(Value::Number(Number::Int(v))) => v,
								_ => return Err("Expected a CBOR integer data type"),
							};

							Utc.timestamp_opt(seconds, nanos)?.into()
						}
```

Reply:
Back and forth with the slow compile time... not really helpful. But I found a better way, using an iterator. Thanks for pointing the unnecessary clone operation.